plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintChecks.kt

```diff
@@ -116,16 +110,15 @@ private fun SolverState.checkDeclarationWithBody(
   declaration: KtDeclaration,
   body: KtExpression?
 ): ContSeq<Unit> =
-  continuationBracket.andThen {
-    val inconsistentPreconditions =
-      checkPreconditionsInconsistencies(constraints, context, declaration)
-    ensure<Unit>(!inconsistentPreconditions)
-    goOn()
-  }.flatMap {
-    checkExpressionConstraints(resultVarName, body, context, ReturnPoints(resultVarName, emptyMap()))
-  }.andThenSideEffect {
-    checkPostConditionsImplication(constraints, context, declaration)
-  }
+    continuationBracket.map {
```

Comment:
Shouldn't this be `flatMap`? Otherwise the call to `ensure` won't influence the flow.

Reply:
Got it! The `map` expects a `ContSyntax`, not a function.